### PR TITLE
Split off pretty-printer package

### PR DESCRIPTION
**Pull request by ChrisPenner**
## Overview

The pretty printer type, pretty printer terminal support, and all the layout stuff is pretty self-contained and rarely changes, so I think it makes sense to split it off 😄.

Helps reduce the monolith that is `parser-typechecker` and should hopefully continue to speed up rebuilds ever so slightly, since the pretty-printer rarely changes.

## Implementation notes

This is just the first minimal step; it just moves the pretty-printer specific modules into a new package.

## Interesting/controversial decisions

`Unison.Lexer.Pos` is used by `AnnotatedText` so it needed to be moved too; despite having `Lexer` in the name it's really just a generic type representing line and column positions so I think it's fine to move out of parser-typechecker. We could probably rename the module to something more generic if we wanted.

## Loose ends

Notably, the following could still use some consideration in the future:

* TermPrinter, TypePrinter, and any utils for pretty-printing domain-specific types are still left inside parser-typechecker
* There are still some spots where we're pretty-printing directly to console in parser-typechecker where we don't even know we're using a cli (e.g. the SqliteCodebase migrations do this)

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     needs:
       - build_linux
       - build_macos
-      - build_windows
```
Comment (ChrisPenner) [inline .github/workflows/release.yaml]: Unless I'm missing something, this diff is broken 🤔 , this isn't what's currently on trunk...

I'll make sure to figure this out before merging.

Comment (aryairani) [reply .github/workflows/release.yaml]: maybe LCA related

File: lib/unison-pretty-printer/tests/Unison/Test/ColorText.hs [hunk 2]
```diff
@@ -7,7 +7,7 @@ module Unison.Test.ColorText where
 import qualified Data.Map as Map
 import EasyTest
 import Text.RawString.QQ
-import Unison.Lexer (Pos (..))
+import Unison.Lexer.Pos (Pos (..))
```
Comment (aryairani) [inline lib/unison-pretty-printer/tests/Unison/Test/ColorText.hs:10]: `Pos` used to be defined in the `Unison.Lexer` module. I split it out to it's own module to eliminate unhelpful dependencies on `Unison.Lexer`, but left a forwarding export there. We can rename the `Pos` module to whatever though, moving forward.

File: lib/unison-pretty-printer/package.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+ghc-options: -Wall -O0 -fno-warn-name-shadowing -fno-warn-missing-pattern-synonym-signatures
+
+flags:
+  optimized:
+    manual: true
+    default: true
+
+when:
+  - condition: flag(optimized)
```
Comment (aryairani) [inline lib/unison-pretty-printer/package.yaml:35]: Need to update `release.yaml` to pass this flag too; or maybe make the default be `-O2` and just use `--fast` to get `-O0`.

**Review by aryairani — APPROVED** [review]

File: lib/unison-pretty-printer/package.yaml (hunk 3 above)
Comment (ChrisPenner) [reply lib/unison-pretty-printer/package.yaml:35]: We currently set the flag on by default in all cases I think, see the lines above, we should probably revisit the way we're setting all these flags at some point, but I think what we have right now should achieve the correct result at least.

```
flags:
  optimized:
    manual: true
    default: true
```

**Pull request merged**